cdap-spark-core-base/src/main/java/io/cdap/cdap/app/runtime/spark/distributed/k8s/SparkContainerDriverLauncher.java

```diff
@@ -0,0 +1,258 @@
+    artifactFetcherService.startAndWait();
+
+    SparkContainerLauncher.launch(delegateClass, delegateArgs.toArray(new String[delegateArgs.size()]));
+  }
+
+
+  private static ArtifactLocalizerClient createArtifactLocalizerClient(CConfiguration cConf, Configuration hConf)
+    throws Exception {
+    MasterEnvironment masterEnv = MasterEnvironments.create(cConf, "k8s");
+    if (masterEnv == null) {
+      throw new RuntimeException("Unable to initialize k8s masterEnv from cConf.");
+    }
+    MasterEnvironmentContext context = MasterEnvironments.createContext(cConf, hConf, masterEnv.getName());
+    masterEnv.initialize(context);
+    MasterEnvironments.setMasterEnvironment(masterEnv);
+
+    Injector injector = createInjector(cConf, hConf, masterEnv);
+    return injector.getInstance(ArtifactLocalizerClient.class);
+  }
+
+  private static Injector createInjector(CConfiguration cConf, Configuration hConf, MasterEnvironment masterEnv) {
+    List<Module> modules = new ArrayList<>();
+
+    CoreSecurityModule coreSecurityModule = CoreSecurityRuntimeModule.getDistributedModule(cConf);
+
+    modules.add(new ConfigModule(cConf, hConf));
+    modules.add(new IOModule());
+    modules.add(new AuthenticationContextModules().getMasterWorkerModule());
+    modules.add(coreSecurityModule);
+
+    modules.add(new AbstractModule() {
+      @Override
+      protected void configure() {
+        bind(DiscoveryService.class)
+          .toProvider(new SupplierProviderBridge<>(masterEnv.getDiscoveryServiceSupplier()));
+        bind(DiscoveryServiceClient.class)
+          .toProvider(new SupplierProviderBridge<>(masterEnv.getDiscoveryServiceClientSupplier()));
+      }
+    });
+    modules.add(new RemoteLogAppenderModule());
+
+    return Guice.createInjector(modules);
+  }
+
+  //TODO (CDAP-18315): following need to be fixed once CDAP-18315 is resolved.
+  private static void setConfigurations(String[] args) throws IOException {
+    Configuration hadoopConf = new Configuration();
+    Properties properties = new Properties();
+    String delegateClass = "org.apache.spark.deploy.SparkSubmit";
+    List<String> delegateArgs = new ArrayList<>();
+    for (int i = 0; i < args.length; i++) {
+      if ("--properties-file".equals(args[i])) {
+        try (InputStream input = new FileInputStream(args[i + 1])) {
+          properties.load(input);
+        }
+      } else if ("--delegate-class".equals(args[i])) {
+        delegateClass = args[i + 1];
+        i++;
+        continue;
+      }
+      delegateArgs.add(args[i]);
+    }
+    for (String property : properties.stringPropertyNames()) {
+      hadoopConf.set(property, properties.getProperty(property));
+    }
+
+    org.apache.hadoop.fs.Path targetDir =
+      new org.apache.hadoop.fs.Path("file:" + new File(WORKING_DIRECTORY).getAbsolutePath());
+    org.apache.hadoop.fs.Path filesPath =
+      new org.apache.hadoop.fs.Path("gs://spark-prototype-masoud/cdap/files/");
```

Comment:
What is this hardcode?

Reply:
These TODOs are required for testing this WIP PR while CDAP-18315 is not complete. 
Won't be submitting the PR with these TODOs.